src/reconstruction.jl

```diff
@@ -0,0 +1,214 @@
+`BC` the boundary condition type and `X` the length of reconstructed vectors.
+Arguments `τ` and `β` are Vectors of `Int` and `CartesianIndex` that contain
+all points to be included in the reconstruction in *relative* coordinates
+and `fsize` is the size of each state in the timeseries.
+
+Note that this is a forward embedding and `τ` needs to be sorted in ascending order.
+The most recent values in the reconstruction are located at the end of each vector.
+
+A simpler constructor for convenience
+
+	SpatioTemporalEmbedding(s, D, τ, B, k, ::Type{<:AbstractBoundaryCondition})
+Takes as arguments the spatial timeseries `s` and reconstructs
+`B` spatial shells separated by `k` points around each point
+and repeats this for `D` past timesteps separated by `τ` each.
+"""
+struct SpatioTemporalEmbedding{T,Φ,BC,X} <: AbstractSpatialEmbedding{T,Φ,BC,X}
+  	τ::Vector{Int}
+	β::Vector{CartesianIndex{Φ}}
+	inner::Region{Φ}  #inner field far from boundary
+	whole::Region{Φ}
+    boundary::BC
+
+	function SpatioTemporalEmbedding{T,Φ,BC,X}(τ,β,bc,fsize) where {T,Φ,BC,X}
+		inner = inner_region(β, fsize)
+		whole = Region((ones(Int,Φ)...,), fsize)
+		return new{T,Φ,BC,X}(τ,β,inner,whole, bc)
+	end
+end
+const STE = SpatioTemporalEmbedding
+
+function SpatioTemporalEmbedding(
+		s::AbstractArray{<:AbstractArray{T,Φ}},
+		D, τ, B, k, boundary::BC
+		) where {T,Φ, BC<:AbstractBoundaryCondition}
+	@assert issorted(τ) "Delays need to be sorted in ascending order"
+    #"ConstantBoundary condition value C needs to be the same type as values in s"
+    @assert BC <: PeriodicBoundary || typeof(boundary.C) == T "typeof(boundary.C) == eltype(s[1])"
+	X = (D+1)*(2B+1)^Φ
+	τs = Vector{Int}(undef,X)
+	βs = Vector{CartesianIndex{Φ}}(undef,X)
+	n = 1
+	for d=0:D, α = Iterators.product([-B*k:k:B*k for φ=1:Φ]...)
+		τs[n] = d*τ
+		βs[n] = CartesianIndex(α)
+		n +=1
+	end
+	return SpatioTemporalEmbedding{T,Φ,BC,X}(τs, βs, boundary, size(s[1]))
+end
+
+
+
+#This function is not safe. If you call it directly with bad params - can fail
+function (r::SpatioTemporalEmbedding{T,Φ,ConstantBoundary{T},X})(rvec,s,t,α) where {T,Φ,X}
+	if α in r.inner
+		@inbounds for n=1:X
+			rvec[n] = s[ t + r.τ[n] ][ α + r.β[n] ]
+		end
+	else
+		@inbounds for n=1:X
+			rvec[n] = α + r.β[n] in r.whole ? s[ t+r.τ[n] ][ α+r.β[n] ] : r.boundary.C
+		end
+	end
+	return nothing
+end
+
+function (r::SpatioTemporalEmbedding{T,Φ,PeriodicBoundary,X})(rvec,s,t,α) where {T,Φ,X}
+	if α in r.inner
+		@inbounds for n=1:X
+			rvec[n] = s[ t + r.τ[n] ][ α + r.β[n] ]
+		end
+	else
+		@inbounds for n=1:X
+			rvec[n] = s[ t + r.τ[n] ][ project_inside(α + r.β[n], r.whole) ]
+		end
+	end
+	return nothing
+end
+
+
+
+get_num_pt(em::SpatioTemporalEmbedding) = prod(em.whole.maxi)
+get_τmax(em::SpatioTemporalEmbedding{T,Φ,BC,X}) where {T,Φ,BC,X} = em.τ[X]
+outdim(em::SpatioTemporalEmbedding{T,Φ,BC,X}) where {T,Φ,BC,X} = X
+
+
+
+
+function Base.summary(::IO, ::SpatioTemporalEmbedding{T,Φ,BC, X}) where {T,Φ,BC,X}
```

Comment:
Overload `show` instead of `summary`? `show` is the method called when you write the object by itself. Not summary.

Also, would be cool for this to print `BC` as well. Also, this is not the correct approach, you have to use `io` inside the print.

So, `Base.show(io, thing) = print(io, "THIS IS MY THING")`